aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -80,6 +80,25 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim) {
 
 // ALL REDUCE #################################################################
 
+static inline Tensor mean(const Tensor &self, optional<ScalarType> dtype) {
+  ScalarType scalarType = self.type().scalarType();
+  AT_CHECK(
+      at::isFloatingType(scalarType),
+      "Can only calculate the average of floating types");
```

Comment:
nit: use the same terminology as the function name, "Can only calculate the mean of floating types".  Also include the dtype string in the message.